### DISCUSSION TITLE
Can we use Mapper on ios devices?

### DISCUSSION THREAD

**Discussion by AhmetPolat43** [OpenOrienteering/mapper#1798] [Questions & Answers] [answered]
Can we use Mapper on ios devices?

**Comment by ghost** [comment]
Duplicate of https://github.com/OpenOrienteering/mapper/issues/1509

TL;DR:  It may be possible to build OpenOrienteering Mapper for iOS, with additional modifications.

But OpenOrienteering Team actually has no plans to provide builds for iOS (iPhone/iPad) devices.

**Reply by MeryS02** [reply]
It may be possible to build OpenOrienteering Mapper for iOS, with additional modifications with AltStore. But how do you do it?

**Comment by lpechacek** [comment]
> But OpenOrienteering Team actually has no plans to provide builds for iOS (iPhone/iPad) devices.

Don't get discouraged by the answer. Mapper is opensource. Feel free to port the program to iOS and contribute the integration bits.

**Comment by dg0yt** [comment, marked as answer]
Just note that is currently not possible to distribute iOS app in Apple's official way.

**Comment by Simon-runner** [comment]
my Mac does not open the Mapper as "it cannot ensure there is no malfare". 
Is there a way to get around this?

**Reply by dg0yt** [reply]
Cf. #2071: If you want to open an app that hasn’t been notarized or is from an unidentified developer: 
https://support.apple.com/en-us/HT202491

(Please note that this discussion here ist for iOS, not macOS.)

**Comment by Simon-runner** [comment]
Vielen Dank, Kai!

-Lauri.


ma 6.6.2022 klo 12.19 Kai Pastor ***@***.***> kirjoitti:

> Cf. #2071 <https://github.com/OpenOrienteering/mapper/issues/2071>: If
> you want to open an app that hasn’t been notarized or is from an
> unidentified developer:
> https://support.apple.com/en-us/HT202491
>
> (Please note that this discussion here ist for iOS, not macOS.)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenOrienteering/mapper/discussions/1798#discussioncomment-2889637>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZP73UQQBIWFSROSBZBU2ODVNW7A7ANCNFSM5X6RGQEQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
-Lauri.

**Comment by sergiogpajares** [comment]
Apple is now allowing app distribution throw new marketplaces [https://support.apple.com/en-us/118110](https://support.apple.com/en-us/118110) like AltStore in the European Union. Would it be possible to include in the roadmap official iPadOS releases? It would be awesome.

**Reply by dg0yt** [reply]
I'm observing these developments. I hope that Apple doesn't add unsurmountable barriers again. (And there are barriers.)

**Reply by acche** [reply]
@dg0yt What obstacles are preventing the launch?

**Comment by acche** [comment]
<img width="1180" height="820" alt="IMG_0493" src="https://github.com/user-attachments/assets/634b9e1c-93f0-4cd6-91cc-e4b87758947a" />
<img width="1180" height="820" alt="IMG_0491" src="https://github.com/user-attachments/assets/8d2ff0df-34fb-4a85-ae38-0a20dc4cff19" />
I ported the mapper to iPad using a LLM - Fable 5, and it runs great.

**Reply by sergiogpajares** [reply]
That’s awesome @acche. Thank you very much. Could you please share source code and binaries? I would like to run it on my own device!

**Reply by acche** [reply]
Thanks, please try it on https://github.com/acche/Mapper

**Reply by Abbe98** [reply]
@acche does it work on iPhones?

**Reply by sergiogpajares** [reply]
I managed to get it working on my iPad. I tested it during real fieldwork, and the application is mostly functional. Thank you very much, @acche!

**Comment by acche** [comment]
Hello maintainers,

I have created an experimental iPadOS port of Mapper:

https://github.com/acche/Mapper/tree/ipados-port

I would like to distribute it free of charge through TestFlight for a small group of testers. The complete source code and build scripts will remain public under GPLv3, with all existing copyright and attribution notices preserved.

Because TestFlight’s terms may impose restrictions that conflict with GPLv3, I would like to ask:

1. Would the copyright holders permit this limited TestFlight distribution?
2. Does the project have authority to grant such permission, or is individual contributor consent required?
3. May I use the OpenOrienteering Mapper name and icon if the build is clearly marked as unofficial?

This is a personal, non-commercial project with no payments, advertising, or in-app purchases. If permission cannot be granted, I will not distribute it publicly through TestFlight.

Thank you for your guidance.

**Reply by dg0yt** [reply]
IANAL.

The copyright is held by the individual contributors. Every contributor would have to agree. In the past, I was not able to reach all copyright holders. This means even I could not get consent.

Also, the app necessarily includes third-party software such as Qt which may require GPL for the combined work. (Things got worse with Qt, and your fork is using Qt6.)

**Reply by acche** [reply]
Then at least a small-scale test should be fine, right?

**Reply by dg0yt** [reply]
Again, IANAL.
This is an old (2010!) FSF post about the topic, but AFAICT nothing has changed, despite some legal pressure on Apple in the EU and elsewhere:
www.fsf.org/blogs/licensing/more-about-the-app-store-gpl-enforcement

> That's the problem in a nutshell: Apple's Terms of Service impose restrictive limits on use and distribution for any software distributed through the App Store, and the GPL doesn't allow that. 

The significant term regard to *your* obligations is "(re)distribution". "Small-scale" testing may or may not constitute "distribution", for the whole aggregate work which forms the app (i.e. including third-party components), and for every "user" which may obtain the app this way.

**Reply by acche** [reply]
I felt like it was getting a bit complicated, so I just went ahead and wrote a version in Swift for iPad.
Hope someone gives it a shot and opens some issues.
Case closed.

**Reply by dg0yt** [reply]
It seems now it is getting complicated.

AFAICS you created a work in AI prompts, not in Swift, and misleadingly label it "OpenOrienteering Mapper for Apple Platforms". Misleading users that way is bad. Stop that.

Being a derived work, it is still covered by GPLv3, so it still incompatible with the app store's term of service.

**Reply by acche** [reply]
This is getting way too complicated. It's completely lost the fun of open source and contributing to orienteering. I'll bury this project and start over.

**Comment by Simon-runner** [comment]
Hi

I love orienteering but these mails are not relevant for me.  How can I
leave the distribution list?

Best
-Lauri.

ti 8.9.2026 klo 18.21 Albin Larsson ***@***.***> kirjoitti:
